File: content_generator/agents/blog_generator.py
"""Blog generator agent."""

from .base_agent import ContentAgent

class BlogGenerator(ContentAgent):
    """Generates blog posts from transcript chunks."""
    
    def generate(self, transcript: str, style_profile: str) -> str:
        """Generate blog post from transcript and style profile.
        
        Args:
            transcript: Content from transcript_chunks.md
            style_profile: Content from style-profile.md
            
        Returns:
            Generated blog post as HTML string
        """
        # Extract style elements and chunks
        style = self._extract_style_elements(style_profile)
        chunks = self._get_chunks(transcript)
        
        # Generate blog post
        blog_post = []
        
        # Add header
        blog_post.append("<article class='blog-post'>")
        blog_post.append("<h1>Finding Your Authentic Path: A Journey of Self-Discovery</h1>")
        
        # Generate sections
        blog_post.extend(self._generate_intro(chunks, style))
        blog_post.extend(self._generate_main_content(chunks, style))
        blog_post.extend(self._generate_conclusion(chunks, style))
        
        # Add footer
        blog_post.append("<section class='cta'>")
        blog_post.append("<h2>Ready to Begin Your Journey?</h2>")
        blog_post.append("<p>Connect with us to explore how you can embrace your authentic self and create lasting impact.</p>")
        blog_post.append("</section>")
        blog_post.append("</article>")
        
        return '\n'.join(blog_post)
    
    def _generate_intro(self, chunks: list[str], style: dict) -> list[str]:
        """Generate blog introduction."""
        intro = []
        intro.append("<section class='intro'>")
        intro.append("<h2>Introduction</h2>")
        intro.append("<p>In a world where everyone seems to have a pre-written script for success, finding your own authentic path can feel like navigating uncharted territory. Today, we explore a personal journey that demonstrates how embracing your unique perspective can lead to genuine impact and meaningful connections.</p>")
        
        # Use first chunk for intro quote
        if chunks:
            paragraphs = chunks[0].strip().split('\n\n')
            if paragraphs:
                intro.append(f"<blockquote>{paragraphs[0].strip()}</blockquote>")
        
        intro.append("</section>")
        return intro
    
    def _generate_main_content(self, chunks: list[str], style: dict) -> list[str]:
        """Generate main blog content."""
        content = []
        content.append("<section class='main-content'>")
        content.append("<h2>The Journey to Authenticity</h2>")
        
        # Process middle chunks
        sections = ['The Turning Point', 'Embracing Your Truth', 'Making an Impact']
        for i, chunk in enumerate(chunks[1:-1], 0):
            section_title = sections[min(i, len(sections)-1)]
            content.append(f"<section class='content-section'>")
            content.append(f"<h3>{section_title}</h3>")
            
            paragraphs = chunk.strip().split('\n\n')
            for p in paragraphs:
                if p.strip():
                    content.append(f"<p>{p.strip()}</p>")
            
            content.append("</section>")
        
        content.append("</section>")
        return content
    
    def _generate_conclusion(self, chunks: list[str], style: dict) -> list[str]:
        """Generate blog conclusion."""
        conclusion = []
        conclusion.append("<section class='conclusion'>")
        conclusion.append("<h2>Embracing Your Path</h2>")
        conclusion.append("<p>The journey to authenticity isn't always straightforward, but it's invariably worthwhile. When we stop trying to fit into someone else's mold and start embracing our unique perspectives, we don't just transform ourselves – we inspire others to do the same.</p>")
        
        # Use last chunk for conclusion
        if chunks:
            paragraphs = chunks[-1].strip().split('\n\n')
            if paragraphs:
                conclusion.append(f"<blockquote>{paragraphs[-1].strip()}</blockquote>")
        
        conclusion.append("</section>")
        return conclusion

File: content_generator/generators/blog_generator.ts
import { writeContent } from '../utils/contentWriter.ts';

interface StyleProfile {
  voice: string[];
  themes: string[];
}

export async function generateBlogPost(transcript: string, styleProfile: StyleProfile) {
  // Extract key insights from the transcript chunks
  const chunks = transcript.split('## [Chunk').slice(1);
  const insights = chunks.map(chunk => {
    const quote = chunk.match(/> Speaker \d+: (.+?)(?=\n|$)/)?.[1] || '';
    return quote.trim();
  }).filter(Boolean);

  // Create blog post structure
  const title = "Finding Your Authentic Path: A Journey of Self-Discovery";
  const introduction = `In a world where everyone seems to have a pre-written script for success, finding your own authentic path can feel like navigating uncharted territory. Today, we explore a personal journey that demonstrates how embracing your unique perspective can lead to genuine impact and meaningful connections.`;

  const mainContent = insights.map(insight => {
    // Transform each insight into a blog section
    return `## ${insight.split('.')[0]}
    
${insight}

${generateReflection(insight, styleProfile)}`;
  }).join('\n\n');

  const conclusion = `The journey to authenticity isn't always straightforward, but it's invariably worthwhile. When we stop trying to fit into someone else's mold and start embracing our unique perspectives, we don't just transform ourselves – we inspire others to do the same.`;

  const blogContent = `# ${title}

${introduction}

${mainContent}

${conclusion}`;

  // Write to both file and database
  await writeContent('output_blog.md', blogContent, {
    section: 'blog',
    title,
    tags: ['personal-growth', 'authenticity', 'leadership', 'self-discovery'],
    excerpt: introduction.split('.')[0] + '.',
    status: 'draft'
  });
  
  return blogContent;
}

function generateReflection(insight: string, styleProfile: StyleProfile): string {
  // Use style profile themes to generate relevant reflections
  const themes = styleProfile.themes || [];
  
  // Map themes to reflective questions and insights
  const reflections = themes
    .filter(theme => insight.toLowerCase().includes(theme.toLowerCase()))
    .map(theme => {
      switch (theme) {
        case 'Personal growth':
          return 'This moment of realization exemplifies how personal growth often comes from challenging our existing beliefs.';
        case 'Finding purpose':
          return 'When we align our actions with our authentic purpose, the path forward becomes clearer.';
        default:
          return 'This insight reflects a deeper truth about our journey toward authenticity and purpose.';
      }
    });

  return reflections[0] || 'This realization opens up new possibilities for growth and connection.';
}
